src/actions/build/mod.rs

```diff
@@ -0,0 +1,140 @@
+mod compiler_message_parser;
+
+use std::collections::HashMap;
+use std::path::{Path, PathBuf};
+use std::sync::{Arc, Mutex};
+
+use analysis::AnalysisHost;
+use hyper::Url;
+use ls_types::Diagnostic;
+use serde_json;
+use span::compiler::DiagnosticSpan;
+
+use build::{BuildPriority, BuildQueue, BuildResult};
+use lsp_data::{NotificationMessage, PublishDiagnosticsParams};
+use server::Output;
+
+use self::compiler_message_parser::{CompilerMessageParser, FileDiagnostic, ParseError};
+
+type Results = HashMap<PathBuf, Vec<Diagnostic>>;
+
+pub struct Build {

```

Comment:
To be clear, I think you are correct about pulling out some of the build-related functions from inside `build`, but I don't think pulling out the data into a `Build` struct is the right approach to doing this.